document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresDatastore.java

```diff
@@ -0,0 +1,120 @@
+package org.hypertrace.core.documentstore.postgres;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.typesafe.config.Config;
+import org.hypertrace.core.documentstore.Collection;
+import org.hypertrace.core.documentstore.Datastore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.*;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class PostgresDatastore implements Datastore {
+  
+  private static final Logger LOGGER = LoggerFactory.getLogger(PostgresDatastore.class);
+  
+  private static final String DEFAULT_DB_NAME = "default_db";
+  Connection client;
+  private String database = DEFAULT_DB_NAME;
+  // Specifies whether document will be stored in json/jsonb format.
+  private String columnType = "json";
+  
+  @Override
+  public boolean init(Config config) {
+    try {
+      DriverManager.registerDriver(new org.postgresql.Driver());
+      if (config.hasPath("type")) {
+        this.columnType = config.getString("type");
+      }
+      // Database needs to be created before initializing connection
+      if (config.hasPath("database")) {
+        this.database = config.getString("database");
+      }
+      String url = String.format("%s/%s", config.getString("url"), database);
+      if (config.hasPath("user") && config.hasPath("password")) {
+        client = DriverManager.getConnection(url, config.getString("user"), config.getString("password"));
+      } else {
+        client = DriverManager.getConnection(url);
+      }
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException(
+        String.format("Unable to instantiate PostgresClient with config:%s", config), e);
+    } catch (SQLException e) {
+      throw new RuntimeException("PostgresClient SQLException", e);
+    }
+    return true;
+  }
+  
+  /**
+   * @return Returns Tables for a particular database
+   */
+  @Override
+  public Set<String> listCollections() {
+    Set<String> collections = new HashSet<>();
+    try {
+      DatabaseMetaData metaData = client.getMetaData();
+      ResultSet tables = metaData.getTables(null, null, "%", new String[]{"TABLE"});
+      while (tables.next()) {
+        collections.add(this.database + "." + tables.getString("TABLE_NAME"));
```

Comment:
Q: will the collection interface impl take care of any table? or It will be restricted to the table described in https://github.com/hypertrace/hypertrace/issues/114 (just having JSONB column)

Reply:
The collection name and column type(json/jsonb) are configurable, but this collection impl will only support the above schema. If there is anything missing in the above schema, we can discuss that. Also, open to making it configurable in any way possible.